### PR TITLE
feat(context): allow URL object on `c.redirect()`

### PR DESCRIPTION
**Pull request by cometkim**
### The author should do the following, if applicable

- [x] Add tests
- [x] Run tests
- [x] `bun run format:fix && bun run lint:fix` to format the code
- [ ] Add [TSDoc](https://tsdoc.org/)/[JSDoc](https://jsdoc.app/about-getting-started) to document the code

### CODE REVIEW

**Comment by EdamAme-x** [comment]
Format check is not complete.  
Do you have different versions of prettier installed globally?

File: src/context.ts [hunk 1]
```diff
@@ -820,11 +820,11 @@ export class Context<
    * ```
    */
   redirect = <T extends RedirectStatusCode = 302>(
-    location: string,
+    location: string | URL,
     status?: T
   ): Response & TypedResponse<undefined, T, 'redirect'> => {
     this.#headers ??= new Headers()
-    this.#headers.set('Location', location)
+    this.#headers.set('Location', location.toString())
```
Comment (EdamAme-x) [inline src/context.ts:827]:
```suggestion
    this.#headers.set('Location', String(location))
```

Although slight, this change reduces the amount of code while maintaining the same operation.

**Review by EdamAme-x — CHANGES_REQUESTED** [review]

**Review by yusukebe — APPROVED** [review]
LGTM!

**Comment by yusukebe** [comment]
Hi @cometkim 

I think this is a good improvement. Looks good to me. Thanks!

**Pull request merged**